### ISSUE TITLE
Does dt_snapshot will influence dt_average? if i set different dt_snapshot,the result of dt_average is different. as the picture shows.

### ISSUE THREAD

**Issue by USTCxiaoxixi** [Warwick-Plasma/epoch#504]
![1](https://github.com/Warwick-Plasma/epoch/assets/133022738/d07d7c5b-6740-4944-bbc5-0c06dc665582)

**Comment by USTCxiaoxixi** [comment]
How to average off a Gaussian laser? Because I am exploring the physical problem, the interaction between laser and matter will produce low-frequency radiation, and in the same time segment, I want to better observe low-frequency radiation, but the presence of reflected laser will cause low-frequency radiation to be not very obvious in the figure, and I try to remove this reflected laser by dt_average.

**Comment by USTCxiaoxixi** [comment]
![1](https://github.com/Warwick-Plasma/epoch/assets/133022738/c4c2b412-cb08-4ebe-a1ed-d33ee9fa61ba)

**Comment by ktangtar** [comment]
The behavior of the averaging is strange when dt_average > dt_snapshot, iirc.  
I don't know if you can average off the gaussian laser because there is both the sinusoidal part and the envelope.  If the envelope is changing very slowly compared to the oscillations, then the averaging off will work better.

**Comment by Status-Mirror** [comment]
I agree with @ktangtar - the envelope amplitude changes too rapidly for the averaging to cancel it out over a small dt_average.

I suspect the best way to remove the gaussian pulse from your figure would be to run a Fourier transform, filter out the laser frequencies, and then perform an inverse Fourier transform. Alternatively, a band-pass filter would do the same trick.

With a larger `dt_average`, you would successfully remove the Gaussian pulse, but you may also smear out your desired signal.

I'll close the issue for now, as I suspect this isn't a bug. Feel free to continue the discussion though below!

Cheers,
Stuart
